### ISSUE TITLE
Releasing a collection and its members isn't releasing all members

### ISSUE THREAD

**Issue by andrewjbtw** [sul-dlss/argo#2243]
**Describe the bug**
Argo supports two types of collection release to SearchWorks:

1. Release only the collection object ("what"="self")
2. Full collection release: release the collection object and its member items ("what"="collection")
Example of a full collection release as shown in the identityMetadata:
```xml
<release what="collection" to="Searchworks" who="lauraw15" when="2020-09-03T23:47:04Z">true</release>
```

The issue is that under some circumstances, full collection release is not releasing the individual items in the collection. Specifically, the cases that have been reported as a problem have followed these steps:

1. Collection is created and items are added to collection
2. Collection is released with "what"="collection"
3. Items do not appear in Searchworks, but collection object does

**User Impact**
Users expect to see the full collection release within the next day but it doesn't happen. Additional steps will be required to get the items to appear in Searchworks.

**To Reproduce**
See above. It's hard to replicate this outside of production because there's no staging Searchworks.

**Expected behavior**
Full collection release should release all of the items together with the collection.

**Additional context**
The full collection release process appears to require the individual item public XML to be republished. This is so that the PURL XML for each collection member gets the proper `releaseData` tags that SearchWorks needs for indexing. The releaseWF has two workflow steps, `release-members` and `release-publish` that probably take care of this. One or both of them may not be working correctly.

If the items in the collection are simply republished (not re-released), Searchworks then picks them up. This suggests the problem is probably is that collection release is not taking care of this publishing step.

Release user documentation is here: https://consul.stanford.edu/display/DLSSDOCS/Managing+the+Release+of+SDR+Resources+to+SearchWorks#ManagingtheReleaseofSDRResourcestoSearchWorks-FullCollectionRelease

**Comment by peetucket** [comment]
Some debugging.  Here is the collection that produced this report (from Slack: https://stanfordlib.slack.com/archives/C0EAZ09QF/p1599239285028400): 

https://argo.stanford.edu/view/druid:kc588th3598

It has many 83 items: 

https://argo.stanford.edu/catalog?f%5Bnonhydrus_collection_title_ssim%5D%5B%5D=Art+Com+records%2C+circa+1970-2005&search_field=text

Now digging through the collection release code, we see the release-members robot is pretty simple:

https://github.com/sul-dlss/common-accessioning/blob/master/lib/robots/dor_repo/release/release_members.rb

It gets the collection, checks to see if the tag is set to "collection" (which means release the whole collection), and then creates the release workflow for each member.

It uses the Dor::Release::MemberService to fetch the members of the collection, via  call to dor-services-app:

https://github.com/sul-dlss/common-accessioning/blob/master/lib/dor/release/member_service.rb#L27-L29

So let's trace this via the common-accessioning console on production:

```
cd common-accessioning
cap prod ssh
ROBOT_ENVIRONMENT=production bin/console
```

And now in ruby-land

```
pid='druid:kc588th3598'
col=Dor::Services::Client.object(pid).find
=> [collection model here]
col.collection?
=> true
col.administrative.releaseTags
=> [#<Cocina::Models::ReleaseTag who="lauraw15" what="collection" date=Thu, 03 Sep 2020 23:47:04 +0000 to="Searchworks" release=true>
robot = Robots::DorRepo::Release::ReleaseMembers.new
robot.send(:'add_wf_to_members?',col)
=> true
```

So far so good.  We have the collection object, it says it's a collection, it has the correct release tag, and the method in the robot that checks to see if we should add the workflow the members evaluates as true.  Now let's try and get the collection members the same way the robot does:

```
member_service = Dor::Release::MemberService.new(druid: druid)
member_service.items
=> []
member_service.send(:members)
=> []
Dor::Services::Client.object(pid).members
=> []
```

That doesn't look right.  The `.items` method uses the ones below it to filter the collection members that are items.  See https://github.com/sul-dlss/common-accessioning/blob/master/lib/dor/release/member_service.rb  And the dor-services-app call is returning nothing.

If this returns no members, then this loop in the robot will do nothing:

https://github.com/sul-dlss/common-accessioning/blob/master/lib/robots/dor_repo/release/release_members.rb#L44-L49

And we will not add the `releaseWF` workflow to any of the items as we should.

So, if this is all correct, the next step is look at dor-services-app and see why the `.members` call isn't working as expected.

**Comment by peetucket** [comment]
Now in dor-services-app, we look at the code that fetches the members of a collection:

https://github.com/sul-dlss/dor-services-app/blob/master/app/controllers/members_controller.rb

It just uses a solr query.  Let's confirm it produces no results:

```
cd dor-services-app
cap prod ssh
bundle exec rails c -e production
```

```
pid='druid:kc588th3598'
solr_params={q: "is_member_of_collection_ssim:\"#{pid}\" published_dttsim:[* TO *]",wt: :json,fl: 'id,objectType_ssim',rows: 100_000_000}
response = ActiveFedora::SolrService.instance.conn.get 'select', params: solr_params
@members = response['response']['docs']
@members.size
 => 0
```

Ok, so the solr query is producing no results.

**Comment by peetucket** [comment]
Here is the solr document for one of the items in the collection:

https://argo.stanford.edu/view/druid:bh354nc2071.json

It does have an `is_member_of_collection_ssim` field, but it has this value: `info:fedora/druid:kc588th3598`.

Let's try to amend the solr query used in dor-services-app to add the `info:fedora` prefix:

```ruby
pid='druid:kc588th3598'
solr_params={q: "is_member_of_collection_ssim:\"info:fedora/#{pid}\" published_dttsim:[* TO *]",wt: :json,fl: 'id,objectType_ssim',rows: 100_000_000}
response = ActiveFedora::SolrService.instance.conn.get 'select', params: solr_params
@members = response['response']['docs']
@members.size
 => 83
```

That looks better.

**Comment by peetucket** [comment]
Found another related issue...

**Comment by peetucket** [comment]
Should be fixed by sul-dlss/dor-services-app#1062 and sul-dlss/dor-services-app/pull/1060  -- will retest when deployed and then close again

**Comment by peetucket** [comment]
Ok, should now be fixed and deployed with changes to dor-services-app and common-accessioning.
